### ISSUE TITLE
after: <class 'pexpect.exceptions.TIMEOUT'>

### ISSUE THREAD

**Issue by iamed2** [xxh/xxh#85]
**Local OS (where xxh is installed):** macOS
**Destination host OS:** Amazon Linux 2
**xxh version:** xxh/0.8.2
**xxh-plugins installed:** fish

**Steps to Reproduce**
1. I run `xxh +vv ocdb-dumper +s fish`
2. The output log:

```
Load xxh config from /Users/ericdavies/.config/xxh/config.xxhc
ssh arguments: ['-o', 'StrictHostKeyChecking=accept-new']
RUN SHELL COMMAND: mkdir -v -p /Users/ericdavies/.xxh /Users/ericdavies/.xxh/.xxh/plugins /Users/ericdavies/.xxh/.xxh/shells
Try pexpect command: bash -c 'echo -e "xxh_home_realpath=\$([ ! -x \"\$(command -v realpath)\" ] && readlink -f ~/.xxh || realpath -m ~/.xxh)\n            xxh_version=\"dir_not_found\"\n            if [[ -d \$xxh_home_realpath ]]; then\n                xxh_version=\$([ \"\$(ls -A \$xxh_home_realpath)\" ] && echo \"version_not_found\" || echo \"dir_empty\")\n                settings_path=\$xxh_home_realpath/.xxh/xxh_version\n                if [[ -f \$settings_path ]]; then\n                    xxh_version=\$(cat \$settings_path)\n                fi\n            fi\n            echo xxh_home_realpath=\$xxh_home_realpath\n            echo xxh_version=\$xxh_version\n            echo xxh_shell_exists=\`[ -d \$xxh_home_realpath/.xxh/shells/xxh-shell-fish ] && echo \"1\" ||echo \"0\"\`\n            echo xxh_home_writable=\`[ -w \$xxh_home_realpath ] && echo \"1\" ||echo \"0\"\`\n            echo xxh_parent_home_writable=\$([ -w \$(dirname \$xxh_home_realpath) ] && echo \"1\" ||echo \"0\")\n            echo rsync=\`command -v rsync\`\n            echo scp=\`command -v scp\`\n            echo shell=\`command -v fish\`\n            echo kernel=\`uname -s\`\n            echo arch=\`uname -m\`" | ssh -v -o StrictHostKeyChecking=accept-new ocdb-dumper -T "bash -s"'
Unknown answer details:
<pexpect.pty_spawn.spawn object at 0x108882c10>
command: /usr/local/bin/bash
args: ['/usr/local/bin/bash', '-c', 'echo -e "xxh_home_realpath=\\$([ ! -x \\"\\$(command -v realpath)\\" ] && readlink -f ~/.xxh || realpath -m ~/.xxh)\\n            xxh_version=\\"dir_not_found\\"\\n            if [[ -d \\$xxh_home_realpath ]]; then\\n                xxh_version=\\$([ \\"\\$(ls -A \\$xxh_home_realpath)\\" ] && echo \\"version_not_found\\" || echo \\"dir_empty\\")\\n                settings_path=\\$xxh_home_realpath/.xxh/xxh_version\\n                if [[ -f \\$settings_path ]]; then\\n                    xxh_version=\\$(cat \\$settings_path)\\n                fi\\n            fi\\n            echo xxh_home_realpath=\\$xxh_home_realpath\\n            echo xxh_version=\\$xxh_version\\n            echo xxh_shell_exists=\\`[ -d \\$xxh_home_realpath/.xxh/shells/xxh-shell-fish ] && echo \\"1\\" ||echo \\"0\\"\\`\\n            echo xxh_home_writable=\\`[ -w \\$xxh_home_realpath ] && echo \\"1\\" ||echo \\"0\\"\\`\\n            echo xxh_parent_home_writable=\\$([ -w \\$(dirname \\$xxh_home_realpath) ] && echo \\"1\\" ||echo \\"0\\")\\n            echo rsync=\\`command -v rsync\\`\\n            echo scp=\\`command -v scp\\`\\n            echo shell=\\`command -v fish\\`\\n            echo kernel=\\`uname -s\\`\\n            echo arch=\\`uname -m\\`" | ssh -v -o StrictHostKeyChecking=accept-new ocdb-dumper -T "bash -s"']
buffer (last 100 chars): b'NAL = iTerm2\r\r\ndebug1: Sending env LC_TERMINAL_VERSION = 3.3.9\r\r\ndebug1: Sending command: bash -s\r\r\n'
before (last 100 chars): b'NAL = iTerm2\r\r\ndebug1: Sending env LC_TERMINAL_VERSION = 3.3.9\r\r\ndebug1: Sending command: bash -s\r\r\n'
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 42386
child_fd: 5
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
Unknown answer from host
Answer from host is empty. Try again with +v or +vv or try ssh before xxh
```

3. What is the problem

For the first few times I used it, it would sometimes return unknown answer and sometimes succeed. The second time I tried, it connected and successfully installed. The fourth time I tried, it connected and I was in a remote Fish session as expected. It did not succeed since then, despite `ssh` still working.

Sometimes the "buffer"/"before" text is different but the rest is consistent.

**For community:**
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by anki-code** [comment]
Hello @iamed2 ! We can see `<class 'pexpect.exceptions.TIMEOUT'>` in the log. This means that connection is slower than expected. 

We increased timeout in 0.8.3 version that will be released soon. But you can try it now from master:
```bash
pip3 install -U git+https://github.com/xxh/xxh
xxh myhost
xxh myhost ++pexpect-timeout 10 # if the TIMEOUT error still appears
```

**Comment by anki-code** [comment]
Closed this because 0.8.3 release should fix this.

**Comment by iamed2** [comment]
I've been behind in my GitHub notifications since my normal app stopped working, but I retested now and it seems to work consistently! Thanks!
